### PR TITLE
Follow up of #2892

### PR DESCRIPTION
**Pull request by rohitgr7**
## What does this PR do?

Follow up of #2892.

# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? 
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by rohitgr7** [comment]
ping @ananyahjha93

File: pytorch_lightning/utilities/data.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+from distutils.version import LooseVersion
+import torch
+from torch.utils.data import DataLoader
+
+from pytorch_lightning.utilities import rank_zero_warn
+
+
+try:
+    from torch.utils.data import IterableDataset
+    ITERABLE_DATASET_EXISTS = True
```
Comment (Borda) [inline pytorch_lightning/utilities/data.py:24]:
```suggestion
    ITERABLE_DATASET_AVALAIBLE = True
```

File: pytorch_lightning/utilities/data.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+except ImportError:
+    ITERABLE_DATASET_EXISTS = False
+
+
+def has_iterable_dataset(dataloader: DataLoader):
+    return ITERABLE_DATASET_EXISTS and hasattr(dataloader, 'dataset') \
+        and isinstance(dataloader.dataset, IterableDataset)
```
Comment (Borda) [inline pytorch_lightning/utilities/data.py:31]:
```suggestion
    return (ITERABLE_DATASET_EXISTS
        and hasattr(dataloader, 'dataset')
        and isinstance(dataloader.dataset, IterableDataset)
    )
```

**Comment by Borda** [comment]
@rohitgr7 is it still WIP?

File: pytorch_lightning/utilities/data.py (hunk 1 above)
Comment (rohitgr7) [reply pytorch_lightning/utilities/data.py]: since min requirements for torch is 1.3, let's just drop this.

**Comment by rohitgr7** [comment]
@Borda good to go now.

**Review by awaelchli — APPROVED** [review]

**Comment by rohitgr7** [comment]
not sure why that test is failing on drone, totally unrelated to this PR. I have seen this problem a lot on the same test. Maybe seed is not set properly for all the tests.

**Comment by awaelchli** [comment]
@rohitgr7 yes, nobody can explain it at this point. I'll look at it this weekend.

**Review by Borda — APPROVED** [review]
